### PR TITLE
Add ArubaOS-CX VSF state sensor support

### PR DESCRIPTION
**Pull request by rudybroersma**
This PR adds sensor support for Aruba VSF (Virtual Switching Framework) sensors:

sample

These sensors are added using PHP code. The reason for that is that the values for these sensors are returned as string (eg 'no_split', or 'ready') and from what I found out it's impossible to have string values in the state indexes. Therefor I added polling translation to convert the possible string values back to numerical values.

```
$ ./lnms dev:check ci --os=arubaos-cx --module=sensors
Only checking os: arubaos-cx
Running unit check... 
Starting snmpsim listening on 127.1.6.2:1162... 
Refreshing database...done
PHPUnit 9.6.10 by Sebastian Bergmann and contributors.

...............................................................  63 / 703 (  8%)
............................................................... 126 / 703 ( 17%)
............................................................... 189 / 703 ( 26%)
............................................................... 252 / 703 ( 35%)
............................................................... 315 / 703 ( 44%)
............................................................... 378 / 703 ( 53%)
............................................................... 441 / 703 ( 62%)
............................................................... 504 / 703 ( 71%)
............................................................... 567 / 703 ( 80%)
............................................................... 630 / 703 ( 89%)
............................................................... 693 / 703 ( 98%)
..........                                                      703 / 703 (100%)

Time: 02:23.972, Memory: 132.00 MB

OK (703 tests, 5847 assertions)
success (160.39s)
```
DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [X] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [ ] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [X] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/polling/sensors/state/arubaos-cx.inc.php [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    //arubaWiredVsfv2MemberTable
+    'not_present' => 10,
+    'booting' => 11,
+    'ready' => 12,
+    'version_mismatch' => 13,
+    'communication_failure' => 14,
+    'in_other_fragment' => 15,
+];
+
+$sensor_value = $stateLookupTable[$sensor_value];
```
Comment (PipoCanaja) [inline includes/polling/sensors/state/arubaos-cx.inc.php:18]: You should here ensure that you are on those 2 OIDs only. Cause if somebody adds another state sensor, this code will break the returned value.

**Review by PipoCanaja — CHANGES_REQUESTED** [review]

File: includes/polling/sensors/state/arubaos-cx.inc.php (hunk 1 above)
Comment (rudybroersma) [reply includes/polling/sensors/state/arubaos-cx.inc.php]: Agreed. I've added an if to check if we're dealing with the VSF sensors.

**Review by PipoCanaja — APPROVED** [review]
LGTM

File: includes/discovery/sensors/state/arubaos-cx.inc.php [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+if (is_array($temp)) {
+    echo 'ArubaOS-CX VSF Operational Status: ';
+    //Create State Index
+    $state_name = 'arubaWiredVsfv2OperStatus';
+    create_state_index($state_name, $vsfOpStatusStates);
+
+    foreach ($temp as $index => $data) {
+        $descr = 'VSF Status';
+        $oid = '.1.3.6.1.4.1.47196.4.1.1.3.15.1.1.1.' . $index;
+        discover_sensor($valid['sensor'], 'state', $device, $oid, $index, $state_name, $descr, 1, 1, null, null, null, null, null, 'snmp', null, null, null, 'VSF');
```
Comment (rudybroersma) [inline includes/discovery/sensors/state/arubaos-cx.inc.php:57]: Here

**Pull request merged**